apm-sniffer/apm-sdk-plugin/dubbo-3.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/asf/dubbo3/DubboInterceptor.java

```diff
@@ -65,15 +63,7 @@ public void beforeMethod(EnhancedInstance objInst, Method method, Object[] allAr
         Invoker invoker = (Invoker) allArguments[0];
         Invocation invocation = (Invocation) allArguments[1];
 
-        RpcServiceContext serviceContext = RpcContext.getServiceContext();
-        boolean isConsumer;
-        URL url = serviceContext.getUrl();
-        if (url == null) {
-            url = serviceContext.getConsumerUrl();
-            isConsumer = url != null;
-        } else {
-            isConsumer = serviceContext.isConsumerSide();
-        }
+        boolean isConsumer = isConsumer(invocation);
```

Comment:
I'm a little confused about this. Why not judge whether it is the consumer side directly from `EnhancedInstance objInst`. If objInst is a MonitorFilter object, it should be on the provider side. If objInst is a MonitorClusterFilter object, it should be on the provider side.

Reply:
We just enhanced `MonitorFilter` class finally, as `MonitorClusterFilter ` extends `MonitorFilter` and enhance both of them will cause problem. We have talked above.

Reply:
@AlbumenJ Agree with @honganan , as a byte code manipulator, once we change the parent class, and the sub classes don't override the changed method, this enhancement actually changes the behavior of all classes on this hierarchy tree.

Reply:
@wu-sheng got it!